src/python/pants/fs/archive.py

```diff
@@ -140,16 +122,15 @@ def _invoke_xz(self, xz_input_file):
     """
     (xz_bin_dir, xz_filename) = split_basename_and_dirname(self._xz_binary_path)
 
-    # TODO(cosmicexplorer): --threads=0 is supposed to use "the number of processor cores on the
-    # machine", but I see no more than 100% cpu used at any point. This seems like it could be a
-    # bug? If performance is an issue, investigate further.
+    # FIXME: --threads=0 is supposed to use "the number of processor cores on the machine", but I
+    # see no more than 100% cpu used at any point. This seems like it could be a bug? If performance
+    # is an issue, investigate further.
     cmd = [xz_filename, '--decompress', '--stdout', '--keep', '--threads=0', xz_input_file]
     env = {
       # Isolate the path so we know we're using our provided version of xz.
       'PATH': xz_bin_dir,
```

Comment:
Rather than setting `$PATH` here, just using `self._xz_binary_path` as the first arg of `cmd` should do the job, right?

Then we can either use the system env, as pants tends to, or an empty env, as I would be very happy about

Reply:
Yes! The `env` bit was only necessary to emulate the kind of wrapping we are now doing in pantsbuild/binaries#72. Were you thinking I might want to explicitly clear the env here?

Reply:
Yeah, explicitly clearing the env in general is nice because it means we're more consistent/reproducible.

It probably doesn't matter either way here, but for things on the build path which we cache (possibly even outside of the local machine) things (like invoking gcc), stripping the entire env, and adding whatever we need explicitly, is a nice safe approach to try to ensure reproducibility. It also means that we can think to explicitly add those env vars to any cache keys we may need to (or, with v2 process execution, this happens by default, so it ensures that we actually get cache hits across users, because otherwise things like `$HOME` will always be different! :))